src/main/java/io/polivakha/mojo/properties/ReadPropertiesMojo.java

```diff
@@ -187,60 +213,38 @@ private void missing( Resource resource )
         }
     }
 
-    private void resolveProperties()
-        throws MojoExecutionException, MojoFailureException
-    {
+    private void resolveProperties() throws MojoExecutionException, MojoFailureException {
         Properties environment = loadSystemEnvironmentPropertiesWhenDefined();
         Properties projectProperties = project.getProperties();
 
-        for ( Enumeration<?> n = projectProperties.propertyNames(); n.hasMoreElements(); )
-        {
-            String k = (String) n.nextElement();
-            projectProperties.setProperty( k, getPropertyValue( k, projectProperties, environment ) );
+        for (Object key : projectProperties.keySet()) {
+            projectProperties.setProperty( (String) key, getPropertyValue( (String) key, projectProperties, environment ) );
```

Comment:
Why didn't you use toString() method on key? 
Maybe it will not in our case and i have big fantasy but
```
            Object uuid = UUID.randomUUID();
            String key1 = (String) uuid;
```
this code brings ClassCastException

Reply:
I do not think that using `toString()` explicitly here makes more sense. Maven project properties are always key value pairs with key as `String` data type. If, for me any reason, bud I'm out coffee base, this will not be a string, then we will indeed get class cast exception. However it is a fall fast approach. The problem with toString is that it is defined in Object, and do it present in every class, so we will be able to invoke it not only on string